### DISCUSSION TITLE
Is its possible to use PyBaMM for single step charge and discharge

### DISCUSSION THREAD

**Discussion by kashifzr** [pybamm-team/PyBaMM#3004] [General]
### Description

I am looking to model lithium ion battery using pybamm package where i am using different value for current to charge and different value for current to discharge in single step while solving or processing the model.
So far what i see in Experiment design, pybamm used single value for charge and discharge with different time span, how to design experiment where charge and discharge works in single step, I know in literature people is saying we cannot do charging and discharging at same time, then How battery in Mobile works where we are charging and discharging with different value while having plug in socket

**Comment by brosaplanella** [comment]
You cannot charge and discharge at the same time (i.e. net current either flows into or out from the battery). It seems that your question is related to an application (how batteries are charged) than to the model itself.
